### ISSUE TITLE
Vim Rendering

### ISSUE THREAD

**Issue by grant-park** [zeit/hyper#770]
Lines will disappear/replicate when I do new line operations in vim. 

Before: https://puu.sh/rvLMZ/3e93f9809e.png
After: https://puu.sh/rvLV2/722b4a8c8c.png

**Comment by bramkok** [comment]
<img width="1143" alt="distorted-vim" src="https://cloud.githubusercontent.com/assets/433068/19026271/eee3dfcc-8923-11e6-980f-14106cfdd192.png">

I have the same behavior when using `hyperterm` with `vim` version 8. However when I start vim without configuration, love and plugins I can't reproduce the behavior.

Please also test with:

```console
vim -u NONE
```

<img width="717" alt="screen shot 2016-10-03 at 03 25 40" src="https://cloud.githubusercontent.com/assets/433068/19026316/65aa34c6-8924-11e6-8924-b2f480896c00.png">

And also no problems when I set  `:syntax on`

<img width="697" alt="screen shot 2016-10-03 at 03 26 02" src="https://cloud.githubusercontent.com/assets/433068/19026326/7dba2c7e-8924-11e6-8364-21f3911c8914.png">

**Comment by bramkok** [comment]
I have similar experiences using `vim` + `hyperterm`. Apparently making a mess of the code. As can be seen in the image below and from my experience: `Terminal.app` and  `iTerm2.app` don't show this type of behavior.

<img width="1143" alt="distorted-vim" src="https://cloud.githubusercontent.com/assets/433068/19025328/cf82b330-8917-11e6-8647-f55e9712cedc.png">

Could you try to reproduce the errors with using `vim` without config, love and plugins?

```console
 vim -u NONE 
```

I've tried this and it seems to solve this problem entirely.

<img width="717" alt="screen shot 2016-10-03 at 03 25 40" src="https://cloud.githubusercontent.com/assets/433068/19025490/c7b14f66-8919-11e6-841a-214a23c48372.png">

Then when I turn on syntax highlighting  (using `:syntax on`) it's again perfectly normal:

<img width="697" alt="screen shot 2016-10-03 at 03 26 02" src="https://cloud.githubusercontent.com/assets/433068/19025503/fd0e1932-8919-11e6-8d08-78e2a53e5945.png">

**Comment by sungjp** [comment]
Can confirm. Thanks for point that out. It's still a shame I can't use my vim plugins/configs...

**Comment by sungjp** [comment]
I did some more playing around and I found that it's just the line numbers that are ruining everything. If I `:set nonumber` then I don't see any rendering issues anymore, even with all my plugins/configs turned on.

**Comment by pardom** [comment]
Seeing the same.

**Comment by tamoore** [comment]
I also see strange rendering issues in VIM 8.0 on macOS Sierra.

I think it might be related to having NERDTree open. When I close NERDTree I don't have the issue anymore.

I've also noticed that problems present themselves when I have an emoji character in the buffer. (I use emoji's to signal syntax or linting problems with syntastic)

<img width="1440" alt="screen shot 2016-12-13 at 8 43 52 pm" src="https://cloud.githubusercontent.com/assets/11851685/21155667/1e895672-c1c7-11e6-988c-dd588be76676.png">

**Comment by mblarsen** [comment]
Note how the `tap` line is slighly mis-aligned because of the emoji in the same line

<img width="440" alt="screenshot 2016-12-14 15 33 55" src="https://cloud.githubusercontent.com/assets/247048/21176111/45bf35e6-c220-11e6-8eb0-f6566e23e498.png">

**Comment by sungjp** [comment]
@tamoore Can you confirm that the problem goes away when you turn off all your plugins (regardless of emoji or no emoji)?

**Comment by tamoore** [comment]
Yeah the problem isn't present when all plugins are disabled. 

Although, having said that, I suspect the issue isn't with either vim or the plugins it seems like a rendering issue with regards to the strange emoji character.

 I tried just copying and pasting an emoji in the prompt to see what effect it would have. It seems that if you paste it then try to add any characters before it (the emoji), it's a little borked. See below:

`<paste> 😀`
<img width="221" alt="screen shot 2016-12-15 at 7 29 17 am" src="https://cloud.githubusercontent.com/assets/11851685/21199601/4a7ce498-c298-11e6-8278-b0b64477115b.png">

`←← '`
<img width="235" alt="screen shot 2016-12-15 at 7 29 24 am" src="https://cloud.githubusercontent.com/assets/11851685/21199654/8475327c-c298-11e6-811b-27d451ddac39.png">

`<delete><delete><delete>`
<img width="242" alt="screen shot 2016-12-15 at 7 29 45 am" src="https://cloud.githubusercontent.com/assets/11851685/21199677/9e2f2268-c298-11e6-956f-e92a26936b7a.png">

**Comment by berkin** [comment]
The unicode chars are wrapped with a span and their width is 1 column. There is a parsing issue i guess, because sometimes i see spaces in the span wrapper of unicode char. As you see in the screenshot, the space after folder icon is inside the span wrapper and folder is sticked to text. 

![image](https://cloud.githubusercontent.com/assets/198913/23986292/7173083c-0a34-11e7-9b65-9094f6da8f62.png)

**Comment by Stanzilla** [comment]
Hey there, please try again with the new Hyper v2 release and open a new issue if the problem stills exists. Thank you!
